core/fortran-example/tutorial/05_parallel_for/CMakeLists.txt

```diff
@@ -0,0 +1,25 @@
+message(${CMAKE_Fortran_COMPILER_ID})
+if (("${CMAKE_Fortran_COMPILER_ID}" STREQUAL "PGI") AND ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "PGI"))
+  if( Cabana_ENABLE_Cuda )
+    add_definitions(-DUSE_GPU=1)
+    set(CMAKE_Fortran_FLAGS "-ta=tesla:cuda9.0 -Minfo=accel -Mcuda=cuda9.0")
+    set(CMAKE_CXX_FLAGS "--relocatable-device-code=true")
+  elseif( Cabana_ENABLE_OpenMP )
+    add_definitions(-DUSE_GPU=0)
+    set(CMAKE_Fortran_FLAGS "-ta=multicore -mp")
+  elseif( Cabana_ENABLE_Serial )
+    add_definitions(-DUSE_GPU=0)
+    set(CMAKE_Fortran_FLAGS "-ta=host")
+  endif()
+  
+  set(CMAKE_EXE_LINKER_FLAGS "-pgc++libs")
```

Comment:
I am not sure how to use it properly. Could you help me with the exact usage here? Thanks!

Reply:
Also, it looks like something new since version 3.14. I am using 3.12.1 and get error message ` Unknown CMake command "target_link_options".` 
If we were to use this new feature, then probably the `cmake_minimum_required(VERSION 3.9)` needs change to 3.14?

Reply:
Suggested by @junghans to write a block of code for version less than 3.13 and version greater than or equal to 3.13